crates/dekaf/src/lib.rs

```diff
@@ -117,13 +134,17 @@ impl SessionAuthentication {
             SessionAuthentication::Task(auth) => {
                 auth.client = auth.client.clone().with_fresh_gazette_client();
             }
+            SessionAuthentication::Redirect { .. } => {
+                unreachable!("This session is a redirect and cannot communicate with Gazette");
```

Comment:
It should be impossible for these `unreachable!` lines to get executed, and if they do then I consider it an implementation bug to be fixed.

I could instead have changed these functions to return `Result<>` and `bail!` here instead, but I didn't.